### ISSUE TITLE
accounts: Group editing

### ISSUE THREAD

**Issue by garrett** [cockpit-project/cockpit#14745]
Accounts page should have group editing, including seeing which users are parts of which group and the ability to add/remove users from group editing.

**Comment by marusak** [comment]
Kind of duplicate of https://github.com/cockpit-project/cockpit/issues/5762

**Comment by marusak** [comment]
> Kind of duplicate of #5762

not it is not :) This is about having view of groups and managing those. Right?

**Comment by garrett** [comment]
This is for editing groups outside of users. It's top-down versus bottom-up.

It's useful for:

- an overview of who is on what group, without having to dive into multiple user groups
- bulk editing (adding several existing users to groups)

The overview is really the most important part, and why I think it'd make sense for a section on the accounts page. (It could even be a sub-page, as having it as a block might just be too much for some organizations.)

**Comment by garrett** [comment]
Additionally, it would also make sense to add command support to groups, outlined in https://www.techrepublic.com/article/how-to-quickly-give-users-sudo-privileges-in-linux/

This would allow people within a certain group to be able to run various commands as root, but not let them do _everything_.

I think we might want to handle all these parts in one big design, but in 3 separate PRs:

1. group overview
   - view-only
   - links from users to groups
   - links from users on the groups to the respective user page
2. group editing, for adding and removing users from groups
3. commands the group is allowed to run
   - for example: all, dnf, docker, etc.

**Comment by andreasn** [comment]
here is the output of less /etc/group on my RHEL 8.2 system at home. Is it common with that many groups? I haven't modified this system _that_ much.
```
root:x:0:
bin:x:1:
daemon:x:2:
sys:x:3:
adm:x:4:
tty:x:5:
disk:x:6:
lp:x:7:
mem:x:8:
kmem:x:9:
wheel:x:10:andreasn
cdrom:x:11:
mail:x:12:
man:x:15:
dialout:x:18:
floppy:x:19:
games:x:20:
tape:x:33:
video:x:39:
ftp:x:50:
lock:x:54:
audio:x:63:
users:x:100:
nobody:x:65534:
dbus:x:81:
utmp:x:22:
utempter:x:35:
input:x:999:
kvm:x:36:
render:x:998:
systemd-journal:x:190:
systemd-coredump:x:997:
systemd-resolve:x:193:
tss:x:59:
polkitd:x:996:
printadmin:x:995:
unbound:x:994:
libstoragemgmt:x:993:
setroubleshoot:x:992:
ssh_keys:x:991:
clevis:x:990:
cockpit-ws:x:989:
sssd:x:988:
sshd:x:74:
chrony:x:987:
slocate:x:21:
tcpdump:x:72:
andreasn:x:1000:
martin:x:1001:
cockpit-wsinstance:x:986:
rngd:x:985:
pcp:x:984:
```

**Comment by marusak** [comment]
> Is it common with that many groups?

yes. But many of them are just user-ones. Like `martin` and `andreasn` are just for your accounts. Also things like `pcp`, `chrony` or `cockpit-wsinstance` are also user ones for users with the same name.

**Comment by garrett** [comment]
Right; this is why I suggested having a couple predefined ones (administrator, container manager, VM manager, and possibly some others that affect Cockpit's UI), some dynamic ones (if it's _used_, but isn't a user's group), then autocomplete for the rest (so they're attainable if someone knows what they are).

If you drop all of the ones that end with ":" in this list, you're left with:

`wheel:x:10:andreasn`

You'd want wheel, but it makes sense to show that as "administrator" in the UI, since wheel doesn't mean anything to anyone other than old UNIX gurus.

**Comment by KKoukiou** [comment]
Do we really want this feature in cockpit? @marusak @garrett @jelly

**Comment by KKoukiou** [comment]
I would say lets close it and if any time in the future someone asks for we we can implement it.

**Comment by marusak** [comment]
What we want on top of what we currently have is ability to add, remove and edit groups.

**Comment by KKoukiou** [comment]
We can now add, rename and edit groups. This issue also request bulk adding/removing of users to groups. Removing/adding users to groups is already possible, per user, in the user details page. Therefore I would consider this very low priority.

**Comment by briansmith0** [comment]
I received feedback that it would be helpful to be able to bulk add/remove users to groups at the group level.  Consider the use case where you'd like to create a new group and add 20 users to the group:  the current work flow requires you to create the group, and then go through and edit each of the 20 user accounts individually to add them to the group.  If you could add/remove users at the group level, this would be significantly faster/easier.
